contracts/test/CompoundGovernorCountingFractional.t.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.26;
+
+import {ProposalTest} from "contracts/test/helpers/ProposalTest.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+import {GovernorCountingFractionalUpgradeable} from "contracts/CompoundGovernor.sol";
+
+abstract contract CountingMode is ProposalTest {
+    function testFuzz_ReturnsCorrectCountingMode() public view {
+        assertEq(governor.COUNTING_MODE(), "support=bravo,fractional&quorum=for,abstain&params=fractional");
+    }
+}
+
+abstract contract HasVoted is ProposalTest {
+    function testFuzz_ReturnsCorrectVotingStatus(
+        bool _hasVotedBefore,
+        uint256 _voteSeed,
+        uint256 _proposerIndex,
+        uint256 _voterIndex
+    ) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        _voterIndex = bound(_voterIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_proposerIndex];
+        address _voter = _majorDelegates[_voterIndex];
+
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _getProposalId(_proposal);
```

Comment:
A general comment here, that probably applies to all of the tests. With the transition to enumerable proposal IDs, it will not be possible to get the proposal ID before it is submitted (unless we add yet another function to the governor to return the "next" proposal ID).. therefore, the call "_getProposalId" will need to be after the call to "_submitProposal()... and the [enumerable proposal ID PR](https://github.com/ScopeLift/compound-governance-upgrade/pull/40) will add functionality to "_getProposalId" to translate the hashed proposal ID to enumerated ID to be returned to the test function.